app/models/request.rb

```diff
@@ -73,6 +73,12 @@ class Request < ApplicationRecord
                 length: { minimum: 1, message: :at_least_one }
       validates :plant_stage, presence: true
     end
+
+    after_transition to: :completed, do: :cleanup_request_distributions
+  end
+
+  def cleanup_request_distributions
+    request_distributions.destroy_all
```

Comment:
Pour ne pas créer une méthode `cleanup_request_distributions` en dehors de la state machine, tu pouvais aussi écrire comme ça, si jamais : 
```suggestion
    after_transition to: :completed do |request, _transition|
      request.request_distributions.destroy_all
    end
  end
```
Mais plus globalement, je me demandais si ça n'était pas dommage de tout supprimer comme ça, on perd le côté historisation ; c'est toujours sympa de pouvoir sauvegarder l'information de "comment cette request a été gérée à l'époque", ça peut toujours servir.

Là en l'état ça marche et je comprends que vous ayez besoin d'aller vite, et qu'une fois qu'une demande est terminée, on vide le bench donc on a plus de distributions, LGTM là dessus. Mais ça pourrait être sympa d'imaginer une évol où on rajoute un état sur les distributions par exemple, ou alors où on les "soft delete" avec discard, quelque chose comme ça :)